lab_6_pipeline/pipeline.py

```diff
@@ -18,21 +49,56 @@ def __init__(self, path_to_raw_txt_data: Path):
         """
         Initializes CorpusManager
         """
+        self._storage = {}
+        self.path = path_to_raw_txt_data
+        self._validate_dataset()
+        self._scan_dataset()
 
     def _validate_dataset(self) -> None:
         """
         Validates folder with assets
         """
+        if not isinstance(self.path, Path) or not self.path.exists():
+            raise FileNotFoundError
+
+        if not self.path.is_dir():
+            raise NotADirectoryError
+
+        if not self.path.glob("*"):
+            raise EmptyDirectoryError
+
+        meta_list = [elem.name for elem in self.path.glob("*.json")]
+        raw_list = [elem.name for elem in self.path.glob("*.txt")]
+        number_list = sorted([int(re.search(r'\d+', elem)[0]) for elem in meta_list])
+        right_list = [i for i in range(1, len(number_list)+1)]
+
+        for element in self.path.glob("*"):
+            if os.stat(element).st_size == 0:
+                raise InconsistentDatasetError
+
+        if len(meta_list) != len(raw_list):
+            raise InconsistentDatasetError
+
+        if number_list != right_list:
```

Comment:
why do not you check numbering of raw files as well?

Reply:
done